0000-overhaul-qiskit-experiments.md

```diff
@@ -0,0 +1,638 @@
+# Overhaul Qiskit Experiments
+
+| **Status**        | **Proposed**                             |
+|:------------------|:-----------------------------------------|
+| **RFC #**         | ####                                     |
+| **Authors**       | Naoki Kanazawa (nkanazawa1989@gmail.com) |
+| **Deprecates**    | N/A                                      |
+| **Submitted**     | 2023-08-23                               |
+| **Updated**       | N/A                                      |
+
+## Summary
+
+[Qiskit Experiments](https://github.com/Qiskit-Extensions/qiskit-experiments) (QE) is a qiskit extension project that provides a basic framework 
+for calibration and characterization experiments together with the built-in experiment library.
+The ultimate goal of this project is the enablement of maintenance workflow for a utility-scale quantum processors, which may consist of 
+few hundreds or maybe thousands of qubits.
+This document spots current performance bottlenecks and describes how to overcome the performance issues.
+
+As described in the [project documentation](https://qiskit.org/ecosystem/experiments/tutorials/intro.html#what-is-qiskit-experiments), QE consists of
```

Comment:
It seem that this RFC handles 2 big proposed changes: 

1. Improving analysis (hence QE) scalability 
2. Breaking `ExperimentData` to more manageable pieces (via executors). 

Obviously 2 facilitates 1 but we can focus on 1 first without going through the steps suggested for 2, if it creates a reasonable overhead with implementing 1 without 2 and then transitioning to 2. So I wonder if this is the case or if doing 1 without 2 is just too much work and these changes should come together

Reply:
Yes, you are correct. 1 and 2 can be done in parallel, and currently @to24toro is investigating the JAX fitter. We can also improve figure generation. However seems like refactoring of composite analysis is very important to improve the user experience. Since composite analysis is tied to the experiment data, so other developer can push forward 2.